documentation/error-handling.md

```diff
@@ -0,0 +1,103 @@
+# Error handling
+
+Because ui extensions script runs in non-browser environment (ex: WebWorker in Web, JavascriptCore on Mobile), it is recommended to listen and send errors to server manually. However, it is still possible to use external libraries for reporting errors. This document shows a couple of examples about how you can handle it.
+
+## The generic way of handling errors
+
+There are two types of errors when using ui extensions: lifecycle and non-lifecycle error.
+
+- **Lifecycle error:** most errors will fall into this type. It could happen when ui extensions script is loaded / imported, the script is run / rendered, or any interaction via callback of ui extensions components or apis. To catch this error, you can add event listener for `unhandledrejection`.
+
+  ```javascript
+  self.onunhandledrejection = (error) => {
+    console.warn('onunhandledrejection', error);
+  };
+  // or
+  self.addEventListener('unhandledrejection', (error) => {
+    console.warn('event unhandledrejection', error);
+  });
+  ```
+
+- **Non-lifecycle error:** this error could happen when the script throw exception while running setTimeout callback. To catch this error, you can add event listener for `error`.
```

Comment:
```suggestion
### Non-lifecycle error

These errors can be thrown when the script encounters an exception while running a `setTimeout` callback. To catch these errors, add an event listener for `error`.
```

Reply:
This is what I observe. Not sure if there are more cases :D